server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java

```diff
@@ -75,8 +76,20 @@ public Map<String, String> getGenerationToPrimaryTermMapper() {
         return generationToPrimaryTermMapper.get();
     }
 
-    public static String getFileName(long primaryTerm, long generation) {
-        return String.join(METADATA_SEPARATOR, Arrays.asList(String.valueOf(primaryTerm), String.valueOf(generation)));
+    /*
+    This should be used only at the time of creation.
+     */
+    public String getFileName() {
+        return String.join(
+            METADATA_SEPARATOR,
+            Arrays.asList(
+                RemoteStoreUtils.invertLong(primaryTerm),
+                RemoteStoreUtils.invertLong(generation),
+                RemoteStoreUtils.invertLong(createdAt),
+                String.valueOf(primaryTerm),
+                String.valueOf(generation)
```

Comment:
Why do we need these when we have the inverted version present?

Reply:
Makes sense. Removing it  in next commit.